benchmarks/sparse/benchmark_cusparselt.py

```diff
@@ -0,0 +1,237 @@
+from itertools import product, combinations
+import torch
+import torch.utils.benchmark as benchmark
+from torch import nn
+from torch.ao.pruning import WeightNormSparsifier
+from tqdm import tqdm
+import pandas as pd
+import argparse
+import random
+from torch.ao.pruning import SemiStructuredSparseTensor
+
+
+torch.set_printoptions(
+    precision=2,
+    threshold=None,
+    edgeitems=16,
+    linewidth=480,
+    profile=None,
+    sci_mode=False,
+)
+
+
+# helper model definition for pruner
+class Model(nn.Module):
+    def __init__(self, m, k, dtype=None):
+        super().__init__()
+        # transposed so reversed
+        self.linear = nn.Linear(k, m)
+
+    def forward(self, x):
+        return self.linear(x)
+
+
+def gen_two_four_sparse_mask(r, c, dtype=torch.float16, device="cuda"):
+    def random_mask_choice(i=None):
+        choices = [
+            [1, 1, 0, 0],
+            [1, 0, 1, 0],
+            [1, 0, 0, 1],
+            [0, 1, 1, 0],
+            [0, 1, 0, 1],
+            [1, 0, 1, 0],
+        ]
+        return choices[random.randint(0, len(choices) - 1) if i is None else i]
+
+    mask_entries = [random_mask_choice() for i in range(r * c // 4)]
+    return (
+        torch.tensor(mask_entries, dtype=dtype, device=device).view(r, c).contiguous()
+    )
+
+
+def test_linear(m, k, n, dtype, contiguous):
+    sparse_weight = torch.rand(m, k).to(dtype).cuda() * gen_two_four_sparse_mask(
+        m, k, dtype=dtype
+    )
+    input_tensor = torch.zeros(n, k).to(dtype).cuda()
+    model = Model(m, k).to(dtype).cuda().eval()
+    model.weight = sparse_weight
+
+    dense_measurement = benchmark.Timer(
+        stmt="model(input_tensor)",
+        globals=locals(),
+    ).blocked_autorange()
+
+    dense_output = model(input_tensor)
+
+    # sparsify weights
+    model.linear.weight = nn.Parameter(
+        SemiStructuredSparseTensor(model.linear.weight, contiguous_output=contiguous)
```

Comment:
Can we have a factory function for this? It's more PyTorch-esque (like torch.ones and torch.empty)